decoder.py

```diff
@@ -46,13 +48,10 @@ def decode_gif(gif_stream: typing.BinaryIO) -> Gif:
                 decode_plain_text(gif_stream, gif_object)
 
         elif prefix is BlockPrefix.ImageDescriptor:
-
-            # Creat a new Image object and add it to the end of the image array in the Gif.
-            gif_object.images.append(Image())
             decode_image_descriptor(gif_stream, gif_object)
 
             # Check if there is a Local color table for this image.
-            if gif_object.images[-1].local_color_table_flag == 1:
+            if gif_object.images[-1].local_color_table_flag:
                 decode_local_color_table(gif_stream, gif_object)
 
             decode_image_data(gif_stream, gif_object)
```

Comment:
@itayshwartz1 @noamTzuber
might be better to add all this logic to decode_image_descriptor